fencing/commands.c

```diff
@@ -124,17 +124,7 @@ static xmlNode *stonith_construct_async_reply(async_command_t * cmd, const char
 static gboolean
 is_action_required(const char *action, stonith_device_t *device)
 {
-    if(device == NULL) {
-        return FALSE;
-
-    } else if (device->required_actions == NULL) {
-        return FALSE;
-
-    } else if (strstr(device->required_actions, action)) {
-        return TRUE;
-    }
-
-    return FALSE;
+    return device && device->automatic_unfencing && safe_str_eq(action, "on");
```

Comment:
I understand you've changed the conditions, but the former format was far more readable